astronomer/providers/utils/sensor_util.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from airflow.exceptions import AirflowException, AirflowSkipException
+from airflow.sensors.base import BaseSensorOperator
+
+if TYPE_CHECKING:  # pragma: no cover
+    from astronomer.providers.utils.typing_compat import Context
+
+
+def poke(cls: BaseSensorOperator, context: Context) -> bool:  # pragma: no cover
+    """Wrapper to call the sensor method"""
+    try:
+        return cls.poke(context)  # type: ignore[return-value]
+    except Exception as e:
+        if cls.soft_fail:
+            raise AirflowSkipException(str(e))
+        else:
+            raise e
+
+
+def raise_error_or_skip_exception(soft_fail: bool, error_message: str) -> None:  # pragma: no cover
```

Comment:
May I know why we mark them as no cover?

Reply:
I thought this not cover by the test but we have a test for this. I have removed it.